x/concentrated-liquidity/incentives_test.go

```diff
@@ -1,23 +1,144 @@
+		MinUptime:       testUptimeFour,
+	}
+
+	testQualifyingDepositsOne   = sdk.NewInt(50)
+	testQualifyingDepositsTwo   = sdk.NewInt(100)
+	testQualifyingDepositsThree = sdk.NewInt(399)
+)
+
+type ExpectedUptimes struct {
+	emptyExpectedAccumValues   []sdk.DecCoins
+	hundredTokensSingleDenom   []sdk.DecCoins
+	hundredTokensMultiDenom    []sdk.DecCoins
+	twoHundredTokensMultiDenom []sdk.DecCoins
+	varyingTokensSingleDenom   []sdk.DecCoins
+	varyingTokensMultiDenom    []sdk.DecCoins
+}
+
+// getExpectedUptimes returns a base set of expected values for testing based on the number
+// of supported uptimes at runtime. This abstraction exists only to ensure backwards-compatibility
+// of incentives-related tests if the supported uptimes are ever changed.
+func getExpectedUptimes() ExpectedUptimes {
+	expUptimes := ExpectedUptimes{
+		emptyExpectedAccumValues:   []sdk.DecCoins{},
+		hundredTokensSingleDenom:   []sdk.DecCoins{},
+		hundredTokensMultiDenom:    []sdk.DecCoins{},
+		twoHundredTokensMultiDenom: []sdk.DecCoins{},
+		varyingTokensSingleDenom:   []sdk.DecCoins{},
+		varyingTokensMultiDenom:    []sdk.DecCoins{},
+	}
+	for i := range types.SupportedUptimes {
+		expUptimes.emptyExpectedAccumValues = append(expUptimes.emptyExpectedAccumValues, cl.EmptyCoins)
+		expUptimes.hundredTokensSingleDenom = append(expUptimes.hundredTokensSingleDenom, sdk.NewDecCoins(cl.HundredFooCoins))
+		expUptimes.hundredTokensMultiDenom = append(expUptimes.hundredTokensMultiDenom, sdk.NewDecCoins(cl.HundredFooCoins, cl.HundredBarCoins))
+		expUptimes.twoHundredTokensMultiDenom = append(expUptimes.twoHundredTokensMultiDenom, sdk.NewDecCoins(cl.HundredFooCoins.Add(cl.HundredFooCoins), cl.HundredBarCoins.Add(cl.HundredBarCoins)))
+		expUptimes.varyingTokensSingleDenom = append(expUptimes.varyingTokensSingleDenom, sdk.NewDecCoins(cl.HundredFooCoins.Add(sdk.NewDecCoin("foo", sdk.NewInt(int64(i))))))
+		expUptimes.varyingTokensMultiDenom = append(expUptimes.varyingTokensMultiDenom, sdk.NewDecCoins(cl.HundredFooCoins.Add(sdk.NewDecCoin("foo", sdk.NewInt(int64(i)))), cl.HundredBarCoins.Add(sdk.NewDecCoin("bar", sdk.NewInt(int64(i*3))))))
+	}
+
+	return expUptimes
+}
+
+// Helper for converting raw DecCoins accum values to pool proto compatible UptimeTrackers
+func wrapUptimeTrackers(accumValues []sdk.DecCoins) []model.UptimeTracker {
+	wrappedUptimeTrackers := []model.UptimeTracker{}
+	for _, accumValue := range accumValues {
+		wrappedUptimeTrackers = append(wrappedUptimeTrackers, model.UptimeTracker{accumValue})
+	}
+
+	return wrappedUptimeTrackers
+}
+
+func expectedIncentives(denom string, rate sdk.Dec, timeElapsed time.Duration, qualifyingLiquidity sdk.Dec) sdk.DecCoin {
+	timeInSec := sdk.NewDec(int64(timeElapsed)).Quo(sdk.MustNewDecFromStr("1000000000"))
+	amount := rate.Mul(timeInSec).Quo(qualifyingLiquidity)
+
+	return sdk.NewDecCoinFromDec(denom, amount)
+}
+
+func chargeIncentive(incentiveRecord types.IncentiveRecord, timeElapsed time.Duration) types.IncentiveRecord {
```

Comment:
I think we can have this helper in prod code, cover it with direct unit tests and then reuse everywhere. WDYT?